acurl/src/request.pyx

```diff
@@ -18,40 +19,42 @@ cdef class Request:
         str url,
         tuple header_tuple,
         tuple cookie_tuple,
-        CURLSH *shared,
         tuple auth,
         bytes data,
         tuple cert,
     ):
         self.method = method
         self.url = url
         self.header_tuple = header_tuple
-        self.cookie_tuple = cookie_tuple
+        self.cookie_tuple = cookie_tuple # Tuple of byte strings
         self.auth = auth
         self.data = data
         self.cert = cert
-        self.session_cookies = self._get_session_cookies(shared)
-        
-    # NOTE: Need to fix the return type
-    cdef tuple _get_session_cookies(self, CURLSH* shared):
+        self.session_cookies = () # Tuple of byte strings
+
+    cdef void store_session_cookies(self, CURLSH* shared):
         cdef CURL* curl = curl_easy_init()
         acurl_easy_setopt_voidptr(curl, CURLOPT_SHARE, shared)
         # NOTE: Do dummy request in order to extract the cookies list
```

Comment:
We don't; see below

Reply:
We have those performance tests right? Would they be sufficient for us to gain confidence that this wouldn't cripple our performance?

Reply:
Yeah, we do have the test suite in `test/perf` in this repo.  (I actually forgot that was there until you mentioned it :sweat_smile:)  I'm not 100% sure that will exercise what we need it to -- it's more focused on the cpu and memory usage of the mite processes (all of them, not just the runner where acurl runs).  IIRC it was originally designed for focusing on the duplicator performance (or perhaps hunting leaks...).

But in any case we can/should definitely run them -- they would be likely to find memory leaks (which I think this code still probably has one or two of, not based on any evidence, but just on the comments about missing destructors in a few places.)  They can also prove that the new acurl doesn't introduce a performance regression in the test, though that's not going to tell us whether the new code is faster or slower than the old, just whether it's "good enough" for this particular test case (which is still more info than we've got now, though).

The tests "should" be runnable very simply by executing the shell script in that directory -- that's the theory anyway, it's been a year+ since they were run so who knows what may happen in practice.  A voyage of discovery :stuck_out_tongue_closed_eyes: